src/cli/main.pl

```diff
@@ -980,50 +986,53 @@
 
     create_authorization(Opts,Authorization),
 
+    format(current_output, "Cloning: origin~n", []),
     api_report_errors(
         clone,
         clone(System_DB, Auth, Organization, DB, Label, Comment, Public, Remote_URL,
               authorized_fetch(Authorization), _Meta_Data)),
-    format(current_output, "~nCloned: ~q into ~s/~s~n", [Remote_URL, Organization, DB]).
+    format(current_output, "Database created: ~s/~s~n", [Organization, DB]).
 run_command(pull,[Path],Opts) :-
     super_user_authority(Auth),
     create_context(system_descriptor{}, System_DB),
-
-    api_report_errors(
-        pull,
-        do_or_die(
-            (   resolve_absolute_string_descriptor(Path,Descriptor),
-                _{ branch_name : Branch} :< Descriptor
-            ),
-            error(not_a_valid_local_branch(Descriptor), _))
-    ),
-
     option(remote(Remote_Name_Atom), Opts),
-
     atom_string(Remote_Name_Atom,Remote_Name),
     option(remote_branch(Remote_Branch), Opts),
-    (   var(Remote_Branch)
-    ->  Branch = Remote_Branch
-    ;   true),
 
     create_authorization(Opts,Authorization),
 
+    format(current_output, "Pulling from: ~s~n", [Remote_Name]),
     api_report_errors(
         pull,
         pull(System_DB, Auth, Path, Remote_Name, Remote_Branch,
              authorized_fetch(Authorization), Result)),
-    format(current_output, "~N~s pulled: ~q~n", [Path, Result]).
+    do_or_die(
+        status{ 'api:pull_status': Pull_Status,
+                'api:fetch_status': Head_Updated } :< Result,
+        error(unexpected_result(pull, Result), _)),
+    (   Pull_Status = "api:pull_unchanged"
+    ->  format(current_output, "Pull unchanged.", [])
+    ;   Pull_Status = "api:pull_fast_forwarded"
+    ->  format(current_output, "Fast-forward pull.", [])
+    ;   format(current_output, "Pull status: ~w.", [Pull_Status])
+    ),
+    (   Head_Updated = true
+    ->  format(current_output, " Branch head updated.", [])
+    ;   Head_Updated = false
+    ->  format(current_output, " Branch head not updated.", [])
+    ;   format(current_output, " Branch head update status: ~w.", [Head_Updated])
+    ).
```

Comment:
Some comments on these messages:
- You're misinterpreting pull_status and fetch_status. fetch_status (which you use for Head_Updated) indicates whether or not we pulled in any changes from the remote. It does not indicate a local branch change. It is pull_status which indicates whether or not a branch changed, and if it didn't change, why it didn't change. It is possible for `fetch_status` to be `true` and `pull_status` to be `pull_unchanged` if none of the remote changes impacted the branch we're pulling into. Likewise, it is possible for `fetch_status` to be `false` and `pull_status` to be something other than `pull_unchanged` if our fetched copy of the repository was already up to date but not yet integrated in our local repository (such as when the user fetches first, or when they previously pulled some other branch triggering that fetch).
- whenever we say we didn't do something, I think it's a good idea to say why we didn't, to prevent users from getting frustrated about something that they feel should have happened but didn't.
- imo messages should include information about what branches and what remotes they operate on.

Reply:
I used  the variable name `Head_Updated` because `src/core/api/db_pull.pl` used `Head_Has_Updated` as a variable for `api:fetch_status`. With the name `api:fetch_status`, I have no idea what the value `true` or `false` means. I believe my mistake is in not realizing it was the _remote_ branch head that was updated, not the _local_ branch head.

I would be happy to provide more information in the messages. Perhaps we need to talk about how to phrase these, because providing the JSON in the CLI messages is not enough.

Reply:
For `Head_Updated`, it's actually neither the remote nor local branch head that was updated. the `Head` in `Head_Updated` refers to the head of the commit graph for the remote repository. It could be that this was updated due to remote changes, but none of these changes are in the branch we're pulling from. So if `true`, the commit graph for the remote repository has a new head (which we just fetched). If `false`, the commit graph for the remote repository was up to date, and nothing was actually downloaded from the remote.

Reply:
Ah, okay. Thanks for the explanation about the commit graph. So, for `api:fetch_status`, we could simply report if the remote commit graph was updated?